src/node_util.cc

```diff
@@ -38,6 +38,11 @@ using v8::Value;
 #undef V
 
 
+static void IsPrimitive(const FunctionCallbackInfo<Value>& args) {
+  args.GetReturnValue().Set(!args[0]->IsObject() && !args[0]->IsExternal());
```

Comment:
There are lots of cases to test it in JS:
```js
const tester = Symbol();
const tested = maybePrimitiveMaybeObject();
tested[tester] = 'Foo';
if(tested[tester]) {
    delete tested[tester]; 
    console.log("Object");
} else {
    console.log("Primitive");
}
```

Or:

```js
var isPrimitive = (function() { return this !== maybePrimitive; }).call(maybePrimitive); // except in strict mode
```

Or without allocating a new object

```js
    var isObject = (typeof maybeObject === "object" && maybeObject !== null) || (typeof maybeObject === "function");
```

Or:

```js
    var isPrimitive = typeof o === "number" || typeof o === "string" || typeof o === "boolean" || typeof o === "undefined" || typeof o === "null" || typeof o === "symbol";
```

Which is why I asked if this is about performance. Is it?